src/app/components/authorization-page/authorization-page.component.ts

```diff
@@ -0,0 +1,88 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { AbstractControl, FormBuilder, FormGroup, ValidationErrors, Validators } from '@angular/forms';
+import { Router } from '@angular/router';
+import { AuthorizationService } from '@services/authorization.service';
+import { Subscription } from 'rxjs';
+
+@Component({
+  selector: 'app-authorization-page',
+  templateUrl: './authorization-page.component.html',
+  styleUrls: ['./authorization-page.component.scss']
+})
+export class AuthorizationPageComponent implements OnInit, OnDestroy {
+
+  public hidePassword = true;
+  public hideRepeatPassword = true;
+  public isRegistration = true;
+  public title = 'Registration';
+  public buttonText = 'Register';
+  public form!: FormGroup;
+  private _subscriptions = new Subscription();
+  private _emailPattern = '^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$';
+  private _passwordPattern = '^(?=.*[0-9])(?=.*[a-z])(?=.*[A-Z])(?=.*?[#?!@$%^&*-]).{8,}$';
+
+  constructor(
+    private formBuilder: FormBuilder,
+    private router: Router,
+    private authService: AuthorizationService
+  ) {
+    if (this.router.url === '/authorization') {
```

Comment:
@KovalivNataliia place this piece of code within a private function and call it here. This will make code more readable